src/OSInfo.cpp

```diff
@@ -182,180 +169,76 @@ QString OSInfo::getMacHash(const QList< QHostAddress > &qlBind) {
+	return QLatin1String("BSD");
+#elif defined(Q_OS_UNIX)
+	return QLatin1String("UNIX");
 #else
-	return QLatin1String("X11");
+	return QLatin1String("Unknown");
 #endif
 }
 
-QString OSInfo::getOSVersion() {
-	static QString qsCached;
-
-	if (!qsCached.isNull())
-		return qsCached.isEmpty() ? QString() : qsCached;
-
-	QString os;
-
+QString OSInfo::getOSDisplayableVersion() {
 #if defined(Q_OS_WIN)
-	OSVERSIONINFOEXW ovi;
-	memset(&ovi, 0, sizeof(ovi));
-
-	ovi.dwOSVersionInfoSize = sizeof(ovi);
-	if (!GetVersionEx(reinterpret_cast< OSVERSIONINFOW * >(&ovi))) {
-		return QString();
-	}
-
-#	if QT_VERSION >= 0x050500
-	os = QString::asprintf("%lu.%lu.%lu.%lu", static_cast< unsigned long >(ovi.dwMajorVersion),
-						   static_cast< unsigned long >(ovi.dwMinorVersion),
-						   static_cast< unsigned long >(ovi.dwBuildNumber),
-						   (ovi.wProductType == VER_NT_WORKSTATION) ? 1UL : 0UL);
-#	else
-	// sprintf() has been deprecated in Qt 5.5 in favor for the static QString::asprintf()
-	os.sprintf("%lu.%lu.%lu.%lu", static_cast< unsigned long >(ovi.dwMajorVersion),
-			   static_cast< unsigned long >(ovi.dwMinorVersion), static_cast< unsigned long >(ovi.dwBuildNumber),
-			   (ovi.wProductType == VER_NT_WORKSTATION) ? 1UL : 0UL);
-#	endif // QT_VERSION >= 0x050500
-
-#elif defined(Q_OS_MAC)
-	SInt32 major, minor, bugfix;
-	OSErr err = Gestalt(gestaltSystemVersionMajor, &major);
-	if (err == noErr)
-		err = Gestalt(gestaltSystemVersionMinor, &minor);
-	if (err == noErr)
-		err = Gestalt(gestaltSystemVersionBugFix, &bugfix);
-	if (err != noErr)
-		return QString::number(QSysInfo::MacintoshVersion, 16);
-
-	char *buildno = nullptr;
-	char buildno_buf[32];
-	size_t sz_buildno_buf = sizeof(buildno);
-	int ret               = sysctlbyname("kern.osversion", buildno_buf, &sz_buildno_buf, nullptr, 0);
-	if (ret == 0) {
-		buildno = &buildno_buf[0];
-	}
-
-#	if QT_VERSION >= 0x050500
-	os = QString::asprintf("%lu.%lu.%lu %s", static_cast< unsigned long >(major), static_cast< unsigned long >(minor),
-						   static_cast< unsigned long >(bugfix), buildno ? buildno : "unknown");
-#	else
-	// sprintf() has been deprecated in Qt 5.5 in favor for the static QString::asprintf()
-	os.sprintf("%lu.%lu.%lu %s", static_cast< unsigned long >(major), static_cast< unsigned long >(minor),
-			   static_cast< unsigned long >(bugfix), buildno ? buildno : "unknown");
-#	endif // QT_VERSION >= 0x050500
-#else
-#	ifdef Q_OS_LINUX
-	QProcess qp;
-	QStringList args;
-	args << QLatin1String("-s");
-	args << QLatin1String("-d");
-	qp.start(QLatin1String("lsb_release"), args);
-	if (qp.waitForFinished(5000)) {
-		os = QString::fromUtf8(qp.readAll()).simplified();
-		if (os.startsWith(QLatin1Char('"')) && os.endsWith(QLatin1Char('"')))
-			os = os.mid(1, os.length() - 2).trimmed();
-	}
-	if (os.isEmpty())
-		qWarning("OSInfo: Failed to execute lsb_release");
-
-	qp.terminate();
-	if (!qp.waitForFinished(1000))
-		qp.kill();
-#	endif
+	auto os = win10DisplayableVersion();
```

Comment:
Please don't use `auto` in this way. It is not clear what type `os` is in this case without having to look up the signature of `win10DisplayableVersion`.